pkg/tbtc/node.go

```diff
@@ -340,12 +340,16 @@ func (n *node) joinDKGIfEligible(seed *big.Int, startBlockNumber uint64) {
 				// perform all the result-signing-related actions and
 				// handle the worst case when the result is submitted by the
 				// last group member.
+				publicationTimeout := time.Duration(chainConfig.GroupSize) *
+					dkgResultSubmissionDelayStep
 				publicationCtx, cancelPublicationCtx := context.WithTimeout(
 					context.Background(),
-					dkgResultSigningDuration+
-						(time.Duration(chainConfig.GroupSize)*dkgResultSubmissionDelayStep),
+					publicationTimeout,
 				)
-				defer cancelPublicationCtx()
+				go func() {
+					defer cancelPublicationCtx()
+					time.Sleep(publicationTimeout)
+				}()
```

Comment:
Do we need this goroutine given that we are constructing `context.WithTimeout`?

Reply:
This addresses https://github.com/keep-network/keep-core/pull/3392#discussion_r1014211031

Reply:
Right, what I mean is that we create context `WithTimeout` which closes the `Done` channel after `publicationTimeout` even without this goroutine. From the docs:

> The returned context's Done channel is closed when the deadline expires, when the returned cancel function is called, or when the parent context's Done channel is closed, whichever happens first.

In other words, this goroutine does the same `context.WithTimeout` already does internally.

In the future, we will cancel the publication context earlier, when the result is published to the chain. So I think we should add a TODO here to avoid confusion. Something like:

 ```
// TODO: Call cancelPublicationCtx() when the result is available
// and published and remove this goroutine.
// This goroutine is duplicating context.WithTimeout work right now.
go func() {
  defer cancelPublicationCtx()
  time.Sleep(publicationTimeout)
}()
```

Alternatively, remove this goroutine and silence the linter warning explaining that the cancel function needs to be called in the future.

Another option would be to, switch this construction to `context.WithCancel` and have this goroutine.

Reply:
Ah got it! Here it is https://github.com/keep-network/keep-core/pull/3392/commits/45ed1f3300373eb6b6c763af406de4edfff5fd51